File: src/test/fixtures/typescript/anonymousCallableAnalysis.ts
const cleanAnonymousCallable = (first: number, second: number): number => first + second;

const flaggedAnonymousCyclomatic = (value: number): number => {
    if (value === 0) return 0;
    if (value === 1) return 1;
    if (value === 2) return 2;
    if (value === 3) return 3;
    if (value === 4) return 4;
    if (value === 5) return 5;
    if (value === 6) return 6;
    if (value === 7) return 7;
    if (value === 8) return 8;
    if (value === 9) return 9;
    if (value === 10) return 10;
    return value;
};

const flaggedSevereAnonymousCyclomatic = (value: number): number => {
    if (value === 0) return 0;
    if (value === 1) return 1;
    if (value === 2) return 2;
    if (value === 3) return 3;
    if (value === 4) return 4;
    if (value === 5) return 5;
    if (value === 6) return 6;
    if (value === 7) return 7;
    if (value === 8) return 8;
    if (value === 9) return 9;
    if (value === 10) return 10;
    if (value === 11) return 11;
    if (value === 12) return 12;
    if (value === 13) return 13;
    if (value === 14) return 14;
    if (value === 15) return 15;
    return value;
};

const flaggedAnonymousCognitive = (value: number): number => {
    if (value > 0) {
        if (value > 1) {
            if (value > 2) {
                if (value > 3) {
                    if (value > 4) {
                        if (value > 5) return value;
                    }
                }
            }
        }
    }
    return 0;
};

const flaggedSevereAnonymousCognitive = (value: number): number => {
    if (value > 0) {
        if (value > 1) {
            if (value > 2) {
                if (value > 3) {
                    if (value > 4) {
                        if (value > 5) {
                            if (value > 6) return value;
                        }
                    }
                }
            }
        }
    }
    return 0;
};

const flaggedAnonymousParameters = function (a: number, b: number, c: number, d: number, e: number, f: number): number {
    return a + b + c + d + e + f;
};

const flaggedSevereAnonymousParameters = (a: number, b: number, c: number, d: number, e: number, f: number, g: number, h: number, i: number): number =>
    a + b + c + d + e + f + g + h + i;

const flaggedAnonymousParameterForms = function (
    a: number,
    b: number = 0,
    c: number = 0,
    d: number = 0,
    e?: number,
    ...rest: number[]
): number {
    return a + b + c + d + (e ?? 0) + rest.length;
};

function nestedCallableBoundary(value: number): number {
    const callback = (item: number): number => item > 0 ? 1 : 0;
    return callback(value);
}

File: src/test/fixtures/fsharp/AnonymousCallableAnalysis.fs
module AnonymousCallableAnalysis

let cleanAnonymousCallable = fun first second -> first + second

let flaggedAnonymousCyclomatic =
    fun value ->
        if value = 0 then 0
        elif value = 1 then 1
        elif value = 2 then 2
        elif value = 3 then 3
        elif value = 4 then 4
        elif value = 5 then 5
        elif value = 6 then 6
        elif value = 7 then 7
        elif value = 8 then 8
        elif value = 9 then 9
        elif value = 10 then 10
        else value

let flaggedSevereAnonymousCyclomatic =
    fun value ->
        if value = 0 then 0
        elif value = 1 then 1
        elif value = 2 then 2
        elif value = 3 then 3
        elif value = 4 then 4
        elif value = 5 then 5
        elif value = 6 then 6
        elif value = 7 then 7
        elif value = 8 then 8
        elif value = 9 then 9
        elif value = 10 then 10
        elif value = 11 then 11
        elif value = 12 then 12
        elif value = 13 then 13
        elif value = 14 then 14
        elif value = 15 then 15
        else value

let flaggedAnonymousCognitive =
    fun value ->
        if value > 0 then
            if value > 1 then
                if value > 2 then
                    if value > 3 then
                        if value > 4 then value else 0
                    else
                        0
                else
                    0
            else
                0
        else
            0

let flaggedSevereAnonymousCognitive =
    fun value ->
        if value > 0 then
            if value > 1 then
                if value > 2 then
                    if value > 3 then
                        if value > 4 then
                            if value > 5 then
                                if value > 6 then value else 0
                            else
                                0
                        else
                            0
                    else
                        0
                else
                    0
            else
                0
        else
            0

let flaggedAnonymousParameters = fun a b c d e f -> a + b + c + d + e + f

let flaggedSevereAnonymousParameters =
    fun a b c d e f g h i -> a + b + c + d + e + f + g + h + i

let nestedCallableBoundary value =
    let callback = fun item -> if item > 0 then 1 else 0
    callback value
